assets/src/presence.ts

```diff
@@ -0,0 +1,97 @@
+type PresenceMetadata = {online_at?: string; phx_ref: string};
```

Comment:
All logic is the same as it was in `ConversationsProvider.tsx`. I just extracted them into this file and created an additional helper method, `updatePresenceWithDiff`, for diffing presence.